client/my-sites/checkout/checkout/source-payment-box.jsx

```diff
@@ -0,0 +1,231 @@
+
+		if ( this.props.selectedSite ) {
+			cancelUrl += this.props.selectedSite.slug;
+		} else {
+			cancelUrl += 'no-site';
+		}
+
+		const dataForApi = {
+			payment: assign( {}, this.state, {
+				paymentMethod: this.paymentMethodByType( this.props.paymentType ),
+				successUrl: origin + this.props.redirectTo(),
+				cancelUrl,
+			} ),
+			cart: this.props.cart,
+			domainDetails: this.props.transaction.domainDetails
+		};
+
+		// get the redirect URL from rest endpoint
+		wpcom.undocumented().transactions( 'POST', dataForApi, function( error, result ) {
+			let errorMessage;
+			if ( error ) {
+				if ( error.message ) {
+					errorMessage = error.message;
+				} else {
+					errorMessage = translate( "We've encountered a problem. Please try again later." );
+				}
+
+				this.setSubmitState( {
+					error: errorMessage,
+					disabled: false
+				} );
+			} else if ( result.redirect_url ) {
+				this.setSubmitState( {
+					info: translate( 'Redirecting you to your bank to complete the payment.' ),
+					disabled: true
+				} );
+				analytics.ga.recordEvent( 'Upgrades', 'Clicked Checkout With Source Payment Button' );
+				analytics.tracks.recordEvent( 'calypso_checkout_with_source_' + this.props.paymentType );
+				location.href = result.redirect_url;
+			}
+		}.bind( this ) );
+	}
+
+	renderButtonText() {
+		if ( cartValues.cartItems.hasRenewalItem( this.props.cart ) ) {
+			return translate( 'Purchase %(price)s subscription with %(paymentProvider)s', {
+				args: { price: this.props.cart.total_cost_display, paymentProvider: this.getPaymentProviderName() }
+			} );
+		}
+
+		return translate( 'Pay %(price)s with %(paymentProvider)s', {
+			args: { price: this.props.cart.total_cost_display, paymentProvider: this.getPaymentProviderName() }
+		} );
+	}
+
+	renderBankOptions() {
+		// Source https://stripe.com/docs/sources/ideal
+		const idealBanks = {
+			abn_amro: 'ABN AMRO',
+			asn_bank: 'ASN Bank',
+			bunq: 'Bunq',
+			ing: 'ING',
+			knab: 'Knab',
+			rabobank: 'Rabobank',
+			regiobank: 'RegioBank',
+			sns_bank: 'SNS Bank',
+			triodos_bank: 'Triodos Bank',
+			van_lanschot: 'Van Lanschot',
+		};
+
+		const idealBanksOptions = map( idealBanks, ( text, optionValue ) => (
+			<option value={ optionValue } key={ optionValue }>{ text }</option>
+		) );
+
+		return [
+			<option value="" key="-">{ translate( 'Please select your bank…' ) }</option>,
+			...idealBanksOptions
+		];
+	}
+
+	render() {
+		const hasBusinessPlanInCart = some( this.props.cart.products, { product_slug: PLAN_BUSINESS } );
+		const showPaymentChatButton =
+			config.isEnabled( 'upgrades/presale-chat' ) &&
+			abtest( 'presaleChatButton' ) === 'showChatButton' &&
+			hasBusinessPlanInCart;
+
+		return (
+			<form onSubmit={ this.redirectToPayment }>
+
+				<div className="checkout__payment-box-section">
+					<Input
+						additionalClasses="checkout-field"
+						name="name"
+						onChange={ this.handleChange }
+						label={ translate( 'Your Name' ) }
```

Comment:
@yoavf If you leave this empty the error message says `Invalid or missing account name` (when we actually just check for "empty"). Should we therefore rename this to `Account Name`?